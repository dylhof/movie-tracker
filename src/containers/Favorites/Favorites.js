import React, { Component } from 'react';
import { connect } from 'react-redux';
import MovieCard from '../MovieCard/MovieCard';
import PropTypes from 'prop-types';

export class Favorites extends Component {
  render() {
    const { movies, favorites } = this.props
    const displayFavorites = movies
      .filter(movie => favorites.includes(movie.id))
      .map(movie => {
        return (
          <MovieCard {...movie} key={movie.id} />
        )
      })

    return (
      <div>
        {!favorites.length ? 
        <div className='add-fave-div'><p className='add-fave-message'>You don't have any favorites yet! Add some to make movie magic!</p></div> : 
        <div className='favorites'>{displayFavorites}</div>}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
  favorites: state.favorites
})

export default connect(mapStateToProps)(Favorites)

Favorites.propTypes = {
  favorites: PropTypes.array,
  movies: PropTypes.array
}