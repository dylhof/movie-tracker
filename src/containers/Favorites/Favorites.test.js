import React from 'react';
import Favorites from './Favorites';
import { shallow } from 'enzyme';
import { mapStateToProps } from './Favorites'

describe('Favorites', () => {
  let wrapper

  it('should match the snapshot with certain favorites', () => {
    //setup
    let mockMovies = [{ title: 'Aquaman', id: 1 }, { title: 'Serenity', id: 2 }]
    let mockFavorites = [2]
    wrapper = shallow(<Favorites
      movies={mockMovies}
      favorites={mockFavorites}
    />)
    //expectation
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with no favorites', () => {
    //setup
    let mockMovies = [{ title: 'Aquaman', id: 1 }, { title: 'Serenity', id: 2 }]
    let mockFavorites = []
    wrapper = shallow(<Favorites
      movies={mockMovies}
      favorites={mockFavorites}
    />)
    //expectation
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {

    it('should return an object with a movies array and favorites array', () => {
      //setup
      const mockState = { movies: [{}], favorites: [1], currentUser: {} }
      const expected = { movies: [{}], favorites: [1] }
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })

})