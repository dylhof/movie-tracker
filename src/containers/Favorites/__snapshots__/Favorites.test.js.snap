// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot with certain favorites 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Favorites should match the snapshot with no favorites 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
